src/components/TreeViewer/hooks/useExpand.js

```diff
@@ -0,0 +1,31 @@
+import { useCallback, useState } from 'react';
+
+export default function useExpand() {
+  const [expandedNodes, setExpandedNodes] = useState([]);
+
+  const togleExpandedNode = useCallback(
+    nodeId => {
+      if (expandedNodes.includes(nodeId)) {
+        setExpandedNodes(expandedNodes.filter(id => id !== nodeId));
+      } else {
+        setExpandedNodes([...expandedNodes, nodeId]);
+      }
+    },
+    [expandedNodes]
```

Comment:
this should be use state functional update like `setState(prev => ... )`, you can avoid `expandedNodes` in deps and avoid re-create function